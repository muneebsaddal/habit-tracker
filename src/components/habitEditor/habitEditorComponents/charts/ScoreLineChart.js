import {
	LineChart,
	Line,
	XAxis,
	YAxis,
	CartesianGrid,
	Tooltip,
	Legend,
	ResponsiveContainer,
} from "recharts";

const ScoreLineChart = ({ data }) => {
	return (
		<ResponsiveContainer width="80%" aspect={4.0 / 2.0} minWidth="300px">
			<LineChart
				data={data}
				margin={{
					top: 5,
					right: 30,
					left: 20,
					bottom: 5,
				}}
			>
				<CartesianGrid strokeDasharray="3 3" />
				<XAxis dataKey="name" />
				<YAxis />
				<Tooltip />
				<Legend />

				<Line
					type="monotone"
					dataKey="score"
					stroke="#4249ff"
					strokeWidth={2}
				/>
			</LineChart>
		</ResponsiveContainer>
	);
};

export default ScoreLineChart;
