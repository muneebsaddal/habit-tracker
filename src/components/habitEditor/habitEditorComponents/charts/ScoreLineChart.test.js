import renderer from "react-test-renderer";
import ScoreLineChart from "./ScoreLineChart";

it("renderers correctly", () => {
	const tree = renderer.create(<ScoreLineChart />).toJSON();
	expect(tree).toMatchSnapshot();
});
