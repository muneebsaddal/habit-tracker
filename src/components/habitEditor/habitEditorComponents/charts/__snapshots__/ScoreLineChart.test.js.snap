// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderers correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "100%",
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": "300px",
      "width": "80%",
    }
  }
/>
`;
